lighthouse-core/gather/gatherers/js-usage.js

```diff
@@ -11,12 +11,29 @@ const Gatherer = require('./gatherer.js');
  * @fileoverview Tracks unused JavaScript
  */
 class JsUsage extends Gatherer {
+  constructor() {
+    super();
+    /** @type {LH.Crdp.Debugger.ScriptParsedEvent[]} */
+    this._scriptParsedEvents = [];
+    this.onScriptParsed = this.onScriptParsed.bind(this);
+  }
+
+  /**
+   * @param {LH.Crdp.Debugger.ScriptParsedEvent} event
+   */
+  onScriptParsed(event) {
+    if (event.embedderName) {
+      this._scriptParsedEvents.push(event);
+    }
+  }
+
   /**
    * @param {LH.Gatherer.PassContext} passContext
    */
   async beforePass(passContext) {
     await passContext.driver.sendCommand('Profiler.enable');
     await passContext.driver.sendCommand('Profiler.startPreciseCoverage', {detailed: false});
+    await passContext.driver.on('Debugger.scriptParsed', this.onScriptParsed);
```

Comment:
send `Debugger.enable`? we're technically covered by driver always doing this, but feels good for clarity